import mysql.connector
from mysql.connector.errors import Error

class MyConnectionError(Exception):
    """Raised when there is a connection"""
    pass

class UseDatabase:
    
    def __init__(self, config: dict) -> None:
        self.configuration = config

    def __enter__(self) -> 'cursor':
        try:
            self.conn = mysql.connector.connect(**self.configuration)
            self.cursor = self.conn.cursor()
            return self.cursor
        except mysql.connector.errors.DatabaseError as err:
            raise MyConnectionError(err)

    def __exit__(self, exc_type, exc_value, exc_trace) -> None:
        self.conn.commit()
        self.cursor.close()
        self.conn.close()
